r/R/expression.R

```diff
@@ -12,7 +12,13 @@
 runExpression <- function(req, data) {
     #Get the annotation matrix with the geneid to name translation, and the subset with the correct names.
     df <- data@misc$gene_annotations
-    genesSubset <- subset(df, toupper(df$name) %in% toupper(req$body$genes))
+    ignore.case <- TRUE
```

Comment:
Why is this set here? Should it be a parameter?

Reply:
Yes, it was originally contemplated in this [thread](https://this-is-biomage.slack.com/archives/C01CP46TJ0Z/p1615819973023100). But it has not yet been decided whether it will be implemented or not.

Reply:
Could you make it a default argument then, to show that the feature can be used at a later date? i.e. `runExpression <- function(req, data, ignore.case=TRUE)`.

Reply:
But, in the case that we add this option in the UI, it should be part of the req. So at the end, the `ignore.case` parameter will belong to `req`.
`req$body$ignore.case`.

Reply:
I personally dislike the idea of a checkbox that the user can deselect so that their search becomes case sensitive. To me it adds clutter to the UI without a clear value (why would the user WANT to have to type the case correctly?).

I think the current behaviour is ok (case insensitive) and that added value for this search will come in the form of gene suggestions as you type.

Reply:
So do you think that it is better to closed the branch?

Reply:
I think so but maybe I'm not thinking of something. What are you're thoughts?

Reply:
Yes, certainly, I don't think there is any use case where it is necessary to switch to sensitive case. Perhaps what might make more sense is to grep by indicating a distance; however that would lead to multiple matches and currently the UI only captures one